### ISSUE TITLE
ANDROID_SDK_ROOT=undefined (recommended setting)

### ISSUE THREAD

**Issue by nicolacardi** [apache/cordova-android#1026]
# Bug Report
I don't know if this is a bug, but it stops me from building my Ionic app

My context:
Mac Catalina 10.15.5
Android Studio 4.0
Ionic 6.10.1
Java 8 update 251 (even though the error say I have JDK version 14!)
Cordova 9.0.0

I run from my terminal:

sudo ionic cordova run android
and after generating some bundles the result is

...
> cordova build android
Checking Java JDK and Android SDK versions
ANDROID_SDK_ROOT=**undefined** (recommended setting)
ANDROID_HOME=/Users/NicolaCardi/Library/Android/sdk (DEPRECATED)
Requirements check failed for JDK 8 ('1.8.*')! Detected version: 14.0.1
Check your ANDROID_SDK_ROOT / JAVA_HOME / PATH environment variables.
[ERROR] An error occurred while running subprocess cordova.
        cordova build android exited with exit code 1.
___________________________________________________________________________________________________

I am using a Mac with zsh shell, so I typed vim ~/.zshrc  in my terminal to edit my .zshrc file (bash_options file is not applicable with zsh) and included:
export ANDROID_HOME=/Users/(myusername)/Library/Android/sdk
export ANDROID_SDK_ROOT=/Users/(myusername/Library/Android/sdk
export ANDROID_AVD_HOME=/Users/(myusername)/.android/avd
export JAVA_HOME=/usr/libexec/java_home
then saved and exited.

Now if I check one by one with an echo, **each of them are fine**, including ANDROID_SDK_ROOT
Nevertheless the build is not completed and that is the message I get.

I have read https://github.com/apache/cordova-android/issues/670 and there they say it's just a "cosmetic issue"....but if it stops a build it doesn't seem to be a cosmetic issue....but a concrete one...

### What is expected to happen?

...to complete the android build

### What does actually happen?

It stops showing that error

## Checklist

- [x] I searched for existing GitHub issues
- [X] I updated all Cordova tooling to most recent version
- [X] I included all the necessary information above

**Comment by breautek** [comment]
Your output contains

> Requirements check failed for JDK 8 ('1.8.*')! Detected version: 14.0.1

It's failing to build because android sdk and tooling needs java 8, but it's finding java14 installation.

**Comment by nicolacardi** [comment]
Thank you breautek for your prompt answer, your suggestion was very helpful and, even though it did not resolve by itself, it put me on the right track: I pointed to the java8 installation as you suggested but still the problem persisted....
With /usr/libexec/java_home -V I realized I have TWO JAVA installations on my Mac and that I simply should **DELETE** the java14 folder and keep only the java 8. Now it builds.
Most probably that double installation caused some confusion to the compiler.
I hope this is useful for somebody else. Thanks again.
